wharehauora-wifi-gateway/wharehauora-wifi-gateway.ino

```diff
@@ -52,14 +73,40 @@ char mqtt_publish_topic[32];
 #define MY_MQTT_PUBLISH_TOPIC_PREFIX mqtt_publish_topic
 #define MY_MQTT_SUBSCRIBE_TOPIC_PREFIX "/sensors/wharehauora/"
 
-
 #define MY_MQTT_USER mqtt_username
 #define MY_MQTT_PASSWORD mqtt_password
 
-
 #include <MySensors.h>
 
+/* ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ */
+
+
+/************************************************************************/
+// Set up heartbeat so we know the gateway is online regardless of
+// whether it has sensors attached
+/* vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv */
 bool shouldSaveConfig = false;
+bool shouldSendHeartbeat = false;
+int heartBeatFrequency = 10; // seconds
+Ticker heartBeat;
+/* ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ */
+
+void debugprintln(String thingToPrint){
+  #ifdef MY_DEBUG
+    Serial.println(thingToPrint);
+  #endif
+}
+void debugprint(String thingToPrint){
+  #ifdef MY_DEBUG
+    Serial.print(thingToPrint);
```

Comment:
There's already `#define MY_DEBUG`
If that's not defined then Serial calls do nothing. (and the memory footprint in the libraries gets real small too)

Reply:
Was it not working?

Reply:
I'll have to double check, I may have just assumed there wasn't though

Reply:
you should be able to just do `Serial.println(thingToPrint);`. If #MY_DEBUG is not true, then it will do nothing and some other magic happens so that the compiled code doesn't include that call.

Reply:
https://github.com/mysensors/MySensors/blob/master/MyConfig.h#L30

```
/**********************************
*  Serial and debug options
***********************************/

// Enable MY_DEBUG in sketch to show debug prints. This option will add a lot to the size of the
// final sketch but is helpful to see what is actually is happening during development
//#define MY_DEBUG

// Enable MY_SPECIAL_DEBUG in sketch to activate I_DEBUG messages if MY_DEBUG is disabled.
// I_DEBUG requests are:
// R: routing info (only repeaters): received msg XXYY (as stream), where XX is the node and YY the routing node
// V: CPU voltage
// F: CPU frequency
// M: free memory
// E: clear MySensors EEPROM area and reboot (i.e. "factory" reset)
//#define MY_SPECIAL_DEBUG

// Enable MY_DEBUG_VERBOSE_SIGNING flag for verbose debug prints related to signing.
// Requires DEBUG to be enabled.
// This will add even more to the size of the final sketch!
//#define MY_DEBUG_VERBOSE_SIGNING

// Enable this in sketch if you want to use TX(1), RX(0) as normal I/O pin
//#define MY_DISABLED_SERIAL

// Enable MY_CORE_ONLY flag if you want to use core functions without loading the framework
//#define MY_CORE_ONLY

// Turn off debug if serial pins is used for other stuff
#ifdef MY_DISABLED_SERIAL
#undef MY_DEBUG
#endif
```

Reply:
cool cool, I'll go get rid of those debugs then.